méli-mélo/2024-04-stepsquiz/js/stepsquiz.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @title WET-BOEW Steps Quiz plugin
+ * @overview Quiz pattern for use with the Steps Form plugin.
+ * @license wet-boew.github.io/wet-boew/License-en.html / wet-boew.github.io/wet-boew/Licence-fr.html
+ * @author ricokola
+ */
+( function( $, document ) {
+"use strict";
+
+//Detect the enhancement of the quiz
+var instances = document.querySelectorAll( ".wb-steps.quiz" );
+
+//How many quiz instances in the page
+instances.forEach ( ( instance ) => {
+	let $instance = $( instance );
+
+	// Calculate number of questions
+	let numQuestion = $( "fieldset", instance ).length;
+
+	// Addition to UI (Ex: progress bar)
+	$( "form", $instance ).prepend( "<p class='progressText' role='status'></p>" );
+	$( "form", $instance ).prepend( "<p><progress class='progressBar' max='" + numQuestion + "'></progress></p>" );
+
+});
+
+var hideOtherSteps = function( e ) {
+	// Get wb-steps component
+	let steps,
+	currentElement = e.currentTarget;
+
+	if ( currentElement.classList.contains( "quiz" ) && currentElement.classList.contains( "wb-steps" ) ) {
+		steps = currentElement;
+
+	} else {
+		steps = $( currentElement ).parentsUntil( ".wb-steps.quiz" ).parent( ).get( 0 );
```

Comment:
Optional: try to only use the JQuery version of the step and use its DOM variant when it does simplify the code, that way it will improve the code readability.

Reply:
A comment has been added to the implementation plan for having the code optimized eventually.